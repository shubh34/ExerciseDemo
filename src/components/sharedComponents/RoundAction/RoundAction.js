import React from 'react';
import PropTypes from 'prop-types';
import './RoundAction.scss';

const RoundAction = (props) => {
	const { element, title } = props;
	return (
		<div className="round-action">
			<button type="button" className="round-action_element"/>
			<span className="round-action_title">{title}</span>
		</div>
	);
};

RoundAction.propTypes = {
	title: PropTypes.string.isRequired,
	element: PropTypes.node.isRequired,
};
export default RoundAction;
