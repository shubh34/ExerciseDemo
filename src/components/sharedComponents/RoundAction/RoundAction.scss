.round-action_element {
    width: 80px;
    height: 80px;
    border-radius: 50%;
    box-shadow: 0 1px 1px rgba(0,0,0,.5);
    background: transparent;
    &:before {
        content: '\e614';
        color: #fff;
        font-family: Glyphs;
        font-size: 40px;
        transition: color .2s ease;
    }
}
