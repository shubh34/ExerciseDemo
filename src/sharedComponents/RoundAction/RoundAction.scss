.round-action {
    display: flex;
    flex-direction: column;
    align-items: center;
}
.round-action_element {
    width: 80px;
    height: 80px;
    border-radius: 50%;
    box-shadow: 0 1px 1px rgba(0,0,0,.5);
    background-color: rgba(0,0,0,.7);
    border: none;
    cursor: pointer;
    &:before {
        content: '\e614';
        color: #fff;
        font-family: Glyphs;
        font-size: 40px;
        transition: color .2s ease;
       
    }
    &:hover {
        &:before {
            color: #ffe300;
        }
    }
    @media screen and (min-width: 1140px) {
        width: 110px;
        height: 110px;
        &:before {
            font-size: 56px;
        }
    }
}
.round-action_title {
    padding: 8px;
}
