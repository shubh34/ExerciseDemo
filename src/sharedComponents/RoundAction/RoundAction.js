import React from 'react';
import PropTypes from 'prop-types';
import './RoundAction.scss';

const RoundAction = (props) => {
	const { title = 'Trailer', onClick } = props;
	return (
		<div className="round-action">
			<button type="button" className="round-action_element" onClick={onClick} />
			<span className="round-action_title">{title}</span>
		</div>
	);
};

RoundAction.propTypes = {
	title: PropTypes.string,
	onClick: PropTypes.func.isRequired,
};
export default RoundAction;
