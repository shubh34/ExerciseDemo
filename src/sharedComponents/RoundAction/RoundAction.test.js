import React from 'react';
import { shallow } from 'enzyme';
import RoundAction from './RoundAction';

describe('RoundAction', () => {
	it('renders without crashing', () => {
		const wrapper = shallow(<RoundAction title="xyz" onClick={jest.fn()} />);
		expect(wrapper).toMatchSnapshot();
	});
	it('should call onClick on button click', () => {
		const onClick = jest.fn();
		const wrapper = shallow(<RoundAction title="xyz" onClick={onClick} />);
		wrapper.find('button').simulate('click');
		expect(onClick).toBeCalled();
	});
});
