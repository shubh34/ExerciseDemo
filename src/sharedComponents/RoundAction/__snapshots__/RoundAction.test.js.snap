// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RoundAction renders without crashing 1`] = `
<div
  className="round-action"
>
  <button
    className="round-action_element"
    name="xyz"
    onClick={[MockFunction]}
    type="button"
  />
  <span
    className="round-action_title"
  >
    xyz
  </span>
</div>
`;
